File: lesson09/app/src/main/java/br/odb/nehe/lesson09/GL2JNILib.java
package br.odb.nehe.lesson09;

import android.content.res.AssetManager;
import android.graphics.Bitmap;

class GL2JNILib {

    public static AssetManager assetManager;

    static {
        System.loadLibrary("NdkGlue");
    }

    /**
     * @param width  the current view width
     * @param height the current view height
     */
    public static native void init(int width, int height);

    public static native void step();

    public static native void tick();

    public static native void onDestroy();

    public static native void onCreate(AssetManager assetManager);

    public static native void setTexture(Bitmap bitmap, Bitmap detail);

    public static native void toggleTwinkling();

    public static native void zoomIn();

    public static native void speedUpTwist();

    public static native void zoomDown();

    public static native void speedDownTwist();

    public static native void reset();
}

File: lesson09/app/src/main/java/br/odb/nehe/lesson09/GL2JNILib.kt
package br.odb.nehe.lesson09

import android.content.res.AssetManager
import android.graphics.Bitmap

internal object GL2JNILib {
	/**
	 * @param width  the current view width
	 * @param height the current view height
	 */
	external fun init(width: Int, height: Int)
	external fun step()
	external fun tick()
	external fun onDestroy()
	external fun onCreate(assetManager: AssetManager?)
	external fun setTexture(bitmap: Bitmap?, detail: Bitmap?)
	external fun toggleTwinkling()
	external fun zoomIn()
	external fun speedUpTwist()
	external fun zoomDown()
	external fun speedDownTwist()
	external fun reset()

	init {
		System.loadLibrary("NdkGlue")
	}
}